File: DailyChallenge/Python/20260109_CircularPrime.py
import math

def is_circular_prime(n):
    numbers = [int(i) for i in str(n)]

    for i in numbers:
        number = int(''.join([str(i) for i in numbers]))

        if not is_prime(number):
            return False

        rotate_array(numbers)
    else:
        return True


def rotate_array(list):
    list.append(list[0])
    list.pop(0)

    return list

def is_prime(n):
    sqr_root = math.sqrt(n)
    
    for i in range(2, math.floor(sqr_root)):
        if n % i == 0:
            return False
    else:
        return True

File: Coding Challenges/Daily Challenge/2026/January/20260109-Circular-Prime.js
function isCircularPrime(n) {
  let numbers = Array.from(n.toString(), Number);

  for(let i = 0; i < numbers.length; i++) {
    let number = Number(numbers.join(''));

    if(!isPrime(number)) {
      return false
    }
	
    rotateArray(numbers)
  }

  return true;
}

function rotateArray(list) {
  list.push(list[0]);
  list.shift()

  return list
}

function isPrime(n) {
  let sqrRoot = Math.sqrt(n)
  
  for(let i = 2; i < sqrRoot; i++) {
    if(n % i == 0) {
      return false;
    }
  }

  return true;
}